tests/std/tests/P0645R10_text_formatting_parse_contexts/test.cpp

```diff
@@ -2,11 +2,108 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <assert.h>
+#include <concepts>
+#include <cstddef>
 #include <format>
+#include <memory>
 #include <string_view>
+#include <type_traits>
 
-// TODO: fill in tests
+using namespace std;
+
+template <class CharType>
+constexpr auto get_input() {
+    if constexpr (same_as<CharType, char>) {
+        return "First {} and second {} and third {}"sv;
+    } else {
+        return L"First {} and second {} and third {}"sv;
+    }
+}
+
+template <class CharType>
+constexpr bool ensure_is_constant_expression(const bool should_be_constant_expression) {
+    basic_format_parse_context<CharType> context{get_input<CharType>(), 1};
+    if (should_be_constant_expression) {
+        context.check_arg_id(0);
+    } else {
+        context.check_arg_id(1);
+    }

```

Comment:
should we go for somethig that implicitly casts to int like `context.check_arg_id(should_be_constant_expression)`

Reply:
Code that implicitly promotes `bool` to `int` is strange - there are a few occurrences in the codebase but I am not super comfortable with them.